src/pages/SunburstPage.tsx

```diff
@@ -0,0 +1,64 @@
+import { sunburstLoader } from "../loaders";
+import Sunburst from "sunburst-chart";
+import { useEffect, useState } from "react";
+import { scaleOrdinal, schemePaired } from "d3";
+
+const SunburstPage = () => {
+  const [data, setData] = useState({});
+  // const data = useLoaderData() as LoaderData<typeof feedbackLoader>;
+  const color = scaleOrdinal(schemePaired);
+
+  useEffect(() => {
+    const loadChartData = async () => {
+      const chartData = await sunburstLoader();
+      setData(chartData);
+    };
+
+    loadChartData();
+  }, []);
```

Comment:
What do you think of using the useLoaderData function, you could get rid of this code by using it and loads the data 

```
 const [data, setData] = useState({});
  // const data = useLoaderData() as LoaderData<typeof feedbackLoader>;
  const color = scaleOrdinal(schemePaired);

  useEffect(() => {
    const loadChartData = async () => {
      const chartData = await sunburstLoader();
      setData(chartData);
    };

    loadChartData();
  }, []);
  ```

**What we are usually doing is fetching on component mount - when data is first needed. That is not great, as it will lead to a loading spinner visible to the user for as long as we are initially fetching.**

**The router however is the first component that always knows which page you are trying to visit, and because it now has loaders, it can know which data those pages need to render.**

Feel free to check this [article](https://tkdodo.eu/blog/react-query-meets-react-router#fetching-early)